pkg/codegen/python/gen_program_expressions.go

```diff
@@ -342,7 +342,7 @@ func (g *generator) GenFunctionCallExpression(w io.Writer, expr *model.FunctionC
 			panic(err)
 		}
 
-		validMethod := makeValidIdentifier(method)
+		validMethod := PyName(method)
```

Comment:
This was a bug in the original Python programgen -- `providerValue` would stay as `providerValue`, when actually it needs to become `provider_value`